Userland/Libraries/LibCrypto/ASN1/DER.cpp

```diff
@@ -52,84 +52,84 @@ Result<size_t, DecodeError> Decoder::read_length()
     return length;
 }
 
-Result<u8, DecodeError> Decoder::read_byte()
+ErrorOr<u8> Decoder::read_byte()
 {
     if (m_stack.is_empty())
-        return DecodeError::NoInput;
+        return Error::from_string_literal("Reading byte from an empty stack");
 
     auto& entry = m_stack.last();
     if (entry.is_empty())
-        return DecodeError::NotEnoughData;
+        return Error::from_string_literal("Reading byte from an empty entry");
 
     auto byte = entry[0];
     entry = entry.slice(1);
 
     return byte;
 }
 
-Result<ReadonlyBytes, DecodeError> Decoder::read_bytes(size_t length)
+ErrorOr<ReadonlyBytes> Decoder::read_bytes(size_t length)
 {
     if (m_stack.is_empty())
-        return DecodeError::NoInput;
+        return Error::from_string_literal("Reading bytes from an empty stack");
```

Comment:
Also, these errors feel worded a bit strange to me, I would expect something like:
`Cannot read bytes from empty stack` but that might just be me.

Reply:
In my opinion this always depends on the context in which the error message ends up. For something like `Runtime Error: <message>`, I feel like what I proposed works a bit better. If someone just dumps the error without context, "Can not..." works better. It's always a bit hard to guess.

Reply:
Yeah, that perfectly reasonable.
Feel free to keep it like this.